docs/search.md

```diff
@@ -38,16 +38,101 @@ curl -XGET 'https://…/api/v1/search' -d '{
+{
+                                "range": {
+                                    "metadata.o2r.temporal.begin": {
+                                        "from": from
+                                    }
+                                }
+                            },
+                            {
+                                "range": {
+                                    "metadata.o2r.temporal.end": {
+                                        "to": to
+                                    }
+                                }
+                            }
+```
+`metadata.o2r.temporal.begin` and `metadata.o2r.temporal.end` are the fields within the dataset which will be queried for the duration which is set by the URL parameters.
 
+#### Parameters for Spatial Searching 
+
+`coords` :- This should cater for the coordinates defined by the bounding box. A 2-d array of coordinates is expected as a parameter. The coordinates are to be 5 pairs of latitude longitude values describing a closed polygon.
+The value from this parameter is used in a pre-configured Elasticsearch spatial query which applies a `geo_shape` filter to fetch the desired ERC according to spatial parameters.
+```bash
+{
+                                "bool": {
+                                    "must": {
+                                        "match_all": {}
+                                    },
+                                    "filter": {
+                                        "geo_shape": {
+                                            "metadata.o2r.spatial.geometry": {
+                                                "shape": {
+                                                    "type": "polygon",
+                                                    "coordinates": coords
+
+
+                                                },
+                                                "relation": "within"
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+```
+The filter has been nested within a `boolean/must match` query so that it completely matches the documents being queried. The filter has been applied to the `metadata.o2r.spatial.geometry` field of the dataset with a `within` relation so that the ERC that matches the document's coordinates completely are fetched.
+
+#### Parameters for Keyword Searching 
+`keyword` :- Any word, phrase in the form of text that should be used to match the title of ERC.
+A simple match query is used at the back-end to match the query value with the values in dataset
+```bash
+{
+    "match" : { "metadata.o2r.title" : keyword}
+  }
+  ```
+  ##### NOTE :
+  The parameters should be used together and not seperatley as they have been coupled together to form one single boolean query as stated below so that a refined search is possible and ERC which passes all the criteria are fetched.
```

Comment:
so the search does not work if one of the URL parameters is missing?

that is unacceptable, the search should also work for only one of temporal, spatial or keyword search.